### PR TITLE
Add ppc cases of cpu mode

### PR DESCRIPTION
**Pull request by chloerh**
Guest can start with different cpu mode setting and do managedsave

Signed-off-by: haizhao <haizhao@redhat.com>

### CODE REVIEW

File: libvirt/tests/src/cpu/ppc_cpu_mode.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        if not status_error:
+            if qemu_line:
+                logging.info('Search for "%s" in qemu command line', qemu_line)
+                qemu_cmd_line = process.run(
+                    'ps -ef|grep qemu', shell=True, verbose=True).stdout_text
+                if qemu_line in qemu_cmd_line:
+                    logging.info('PASS: qemu command line check')
+                else:
+                    test.fail('Fail: qemu command line check')
+            managed_save_result = virsh.managedsave(vm_name, debug=True)
```
Comment (sathnaga) [inline libvirt/tests/src/cpu/ppc_cpu_mode.py]: could we make it generic as event and we can control it from config whichever event we want, save/restore, suspend/resume, etc...

Comment (chloerh) [reply libvirt/tests/src/cpu/ppc_cpu_mode.py]: That's brilliant! Updated.

**Comment by chloerh** [comment]
Test result on Power9
---

negative_test.custom.match.none.model.default.power42 | 17 秒 | Passed
negative_test.custom.match.val.exact.model.fallback.power7 | 16 秒 | Fixed
negative_test.host_model.match.none.model.default.power42 | 16 秒 | Passed
positive_test.custom.match.val.exact.model.fallback.power8 | 0 毫秒 | Skipped
positive_test.custom.match.val.exact.model.fallback.power9 | 23 秒 | Passed
positive_test.custom.match.val.minimum.model.fallback.power8 | 0 毫秒 | Skipped
positive_test.custom.match.val.minimum.model.fallback.power9 | 21 秒 | Passed
positive_test.custom.match.val.strict.model.fallback.power8 | 0 毫秒 | Skipped
positive_test.custom.match.val.strict.model.fallback.power9 | 26 秒 | Passed
positive_test.host_model.match.none.model.fallback.power7 | 23 秒 | Fixed
positive_test.host_model.match.none.model.fallback.power8 | 0 毫秒 | Skipped
positive_test.host_model.match.none.model.fallback.power9 | 22 秒 | Passed
positive_test.host_model.match.none.no_model | 24 秒 | Passed
positive_test.host_passthru.match.none.no_model | 23 秒 | Passed

File: libvirt/tests/cfg/cpu/ppc_cpu_mode.cfg [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+            cpu_mode = host-passthrough
+            only match.none.no_model
+    variants:
+        - positive_test:
+            no model.default, custom.match.none, custom..no_model, custom..power7
+            status_error = no
+            no_model: qemu_line = -cpu host
+        - negative_test:
+            status_error = yes
+            only model.default, custom..exact..power7
```
Comment (kylazhang) [inline libvirt/tests/cfg/cpu/ppc_cpu_mode.cfg]: do we need "only pseries" ?

File: libvirt/tests/cfg/cpu/ppc_cpu_mode.cfg [hunk 3]
```diff
@@ -0,0 +1,59 @@
+- ppc_cpu_mode:
+    type = ppc_cpu_mode
+    start_vm = no
+    status_error = no
+    do_managedsave = yes
+    variants:
+        - no_model:
+            qemu_line = -cpu host
```
[the remark this answers is not in the document]
Comment (chloerh) [reply libvirt/tests/cfg/cpu/ppc_cpu_mode.cfg]: Updated

File: libvirt/tests/cfg/cpu/ppc_cpu_mode.cfg (hunk 2 above)
Comment (chloerh) [reply libvirt/tests/cfg/cpu/ppc_cpu_mode.cfg]: Add to line:6 with #

**Pull request merged**